### DISCUSSION TITLE
Object and object.field isn't updating one another

### DISCUSSION THREAD

**Discussion by ap-justin** [react-hook-form/react-hook-form#9273] [Q&A] [answered]
I have a typical token amount with currency selector component - [see sandbox](https://codesandbox.io/s/object-v-field-0f26m2?file=/src/App.tsx)
where the dropdown selects a `token` and the `text input` modifies `token.amount`
![image](https://user-images.githubusercontent.com/89639563/198186956-7a7eb8cd-1056-4ea0-8d99-4d0e6fd80e6f.png)

expected behavior: 
1. modify text input to `123`
2. select another `token`
3.  input should reset to `""`

but `text-input` still is `123`

I hope someone could tell me what I'm missing or do not understand 🙏

**Comment by leapful** [comment, marked as answer]
@ap-justin : Your implementation is using object `token` as a field value and its nested field `token.amount` as a field value too, this is not the correct way to handle form values with react-hook-form because each field registered with react-hook-form will be handled as a separated reference therefore when `token` changed, it does not trigger change on `token.amount`.

You should separate input `token.amount` and listbox `token.symbol` then reset `token.amount` by using method `setValue` when `token.symbol` changed.

Working demo: https://codesandbox.io/s/object-v-field-forked-xxdk6c?file=/src/App.tsx

**Reply by ap-justin** [reply]
@leapful, your idea is perfect! thank you for sharing this to me - much appreciated
